profiling/src/php_ffi.c

```diff
@@ -324,14 +331,22 @@ void ddog_php_prof_function_run_time_cache_init(const char *module_name) {
     // Grab 1 slot for caching filename, as it turns out the utf-8 validity
     // check is worth caching.
 #if PHP_VERSION_ID < 80200
-    ddog_php_prof_run_time_cache_handle =
+    _user_run_time_cache_handle =
         zend_get_op_array_extension_handle(module_name);
     int second = zend_get_op_array_extension_handle(module_name);
-    ZEND_ASSERT(ddog_php_prof_run_time_cache_handle + 1 == second);
+    ZEND_ASSERT(_user_run_time_cache_handle + 1 == second);
 #else
-    ddog_php_prof_run_time_cache_handle =
+    _user_run_time_cache_handle =
+        zend_get_op_array_extension_handles(module_name, 2);
+
+#if PHP_VERSION_ID >= 80400
+    // On PHP 8.4+, the internal cache slots need to be registered separately
+    // from the user ones.
+    _internal_run_time_cache_handle =
         zend_get_op_array_extension_handles(module_name, 2);
```

Comment:
You meant to call zend_get_internal_function_extension_handle here? You're calling op_array again.